### PR TITLE
chore(feat): add error convention and implementing on error hook

### PR DESCRIPTION
**Pull request by ikr4-m**
This error convention based on [IETF RFC 9457](https://datatracker.ietf.org/doc/html/rfc9457) and it will introduce on Agni tutorial.

### CODE REVIEW

**Review by KagChi — APPROVED** [review]

File: src/Function/OnNotFound.ts [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+import ErrorBuilder from "App/Error/ErrorBuilder.js";
 import isJson from "App/Function/IsJson.js";
 
 export function onNotFound(c: Context<any, any, any>): Response {
-    const message = "Path not found.";
     const code = 404;
+    const builder = new ErrorBuilder("Not found.", "Path or route not found.");
+    builder
+        .setInstance(c.req.path)
+        .setType(`https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/${code}`);
```
Comment (satoufuyuki) [inline src/Function/OnNotFound.ts]: If you are planning to use this MDN link as a type to explain HTTP code, I would suggest moving the link to some constant or a helper function.

Comment (ikr4-m) [reply src/Function/OnNotFound.ts]: Thanks for the suggestion. I think that's a good idea to make some method to generate type from MDN link. I'll try to implement it on the codebase.

Comment (ikr4-m) [reply src/Function/OnNotFound.ts]: Please check on this code.
https://github.com/kakushindev/agni/blob/fff47944bec5a1ec54df9d4390f8d0a43e7d3b5f/src/App/Error/ErrorBuilder.ts#L42-L47
https://github.com/kakushindev/agni/blob/fff47944bec5a1ec54df9d4390f8d0a43e7d3b5f/src/Function/OnNotFound.ts#L8-L10
https://github.com/kakushindev/agni/blob/fff47944bec5a1ec54df9d4390f8d0a43e7d3b5f/src/Function/OnValidateError.ts#L9-L12

**Review by satoufuyuki — APPROVED** [review]
LGTM 👍

**Pull request merged**